avm/res/machine-learning-services/workspace/main.bicep

```diff
@@ -0,0 +1,477 @@
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource cMKKeyVault 'Microsoft.KeyVault/vaults@2023-02-01' existing = if (!empty(customerManagedKey.?keyVaultResourceId)) {
+  name: last(split((customerManagedKey.?keyVaultResourceId ?? 'dummyVault'), '/'))
+  scope: resourceGroup(split((customerManagedKey.?keyVaultResourceId ?? '//'), '/')[2], split((customerManagedKey.?keyVaultResourceId ?? '////'), '/')[4])
+
+  resource cMKKey 'keys@2023-02-01' existing = if (!empty(customerManagedKey.?keyVaultResourceId) && !empty(customerManagedKey.?keyName)) {
+    name: customerManagedKey.?keyName ?? 'dummyKey'
+  }
+}
+
+resource cMKUserAssignedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = if (!empty(customerManagedKey.?userAssignedIdentityResourceId)) {
+  name: last(split(customerManagedKey.?userAssignedIdentityResourceId ?? 'dummyMsi', '/'))
+  scope: resourceGroup(split((customerManagedKey.?userAssignedIdentityResourceId ?? '//'), '/')[2], split((customerManagedKey.?userAssignedIdentityResourceId ?? '////'), '/')[4])
+}
+
+resource workspace 'Microsoft.MachineLearningServices/workspaces@2022-10-01' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: sku
+    tier: sku
+  }
+  identity: identity
+  properties: {
+    friendlyName: name
+    storageAccount: associatedStorageAccountResourceId
+    keyVault: associatedKeyVaultResourceId
+    applicationInsights: associatedApplicationInsightsResourceId
+    containerRegistry: associatedContainerRegistryResourceId
+    hbiWorkspace: hbiWorkspace
+    allowPublicAccessWhenBehindVnet: allowPublicAccessWhenBehindVnet
+    description: description
+    discoveryUrl: discoveryUrl
+    encryption: !empty(customerManagedKey) ? {
+      status: 'Enabled'
+      identity: !empty(customerManagedKey.?userAssignedIdentityResourceId) ? {
+        userAssignedIdentity: cMKUserAssignedIdentity.id
+      } : null
+      keyVaultProperties: {
+        keyVaultArmId: cMKKeyVault.id
+        keyIdentifier: !empty(customerManagedKey.?keyVersion ?? '') ? '${cMKKeyVault::cMKKey.properties.keyUri}/${customerManagedKey!.keyVersion}' : cMKKeyVault::cMKKey.properties.keyUriWithVersion
+      }
+    } : null
+    imageBuildCompute: imageBuildCompute
+    primaryUserAssignedIdentity: primaryUserAssignedIdentity
+    publicNetworkAccess: !empty(publicNetworkAccess) ? any(publicNetworkAccess) : (!empty(privateEndpoints) ? 'Disabled' : 'Enabled')
+    serviceManagedResourcesSettings: serviceManagedResourcesSettings
+    sharedPrivateLinkResources: sharedPrivateLinkResources // Note: This property is not idempotent. Neither with [] or `null`
```

Comment:
I wonder if that comment should rather be in the parameter's description? 😄

Reply:
Moved it there: [86efb8b](https://github.com/Azure/bicep-registry-modules/pull/1055/commits/86efb8bf26f57ae8d6a923b595adea3a9b8e536c)